torch/csrc/jit/passes/batch_mm.cpp

```diff
@@ -111,8 +124,25 @@ RegisterOperators mm_tree_reduction_reg(
         // failing
         if (have_same_shape(lhs_inputs) && have_same_shape(rhs_inputs) &&
             shape_is_fast_for_reduce(lhs_inputs[0], rhs_inputs[0])) {
-          auto lhs = at::cat(lhs_inputs, /*dim=*/1);
-          auto rhs = at::cat(rhs_inputs, /*dim=*/0);
+          //sometimes lhs_inputs or rhs_inputs are not contiguous, and that causes at::cat to go through slow path
+          //view them as contiguous if possible by transposing
+          bool lhs_input_transposed = (!lhs_inputs[0].is_contiguous() && lhs_inputs[0].t().is_contiguous());
```

Comment:
Since we know that they're 2D can't we just do the stride check manually? `.t().is_contiguous()` is quite convenient, but it allocates a whole new tensor which is a total overkill for this case.